File: blueteeth/src/main/kotlin/com/robotpajamas/blueteeth/models/Logger.kt
package com.robotpajamas.blueteeth.models

import com.robotpajamas.blueteeth.blueteethLogger

interface Logger {
    fun verbose(message: String)
    fun debug(message: String)
    fun info(message: String)
    fun warning(message: String)
    fun error(message: String)
}

object BLog {
    fun v(message: String, vararg args: Any?, tag: String = "Blueteeth") {
        blueteethLogger?.verbose(String.format("$tag: $message", args))
    }

    fun d(message: String, vararg args: Any?, tag: String = "Blueteeth") {
        blueteethLogger?.debug(String.format("$tag: $message", args))
    }

    fun i(message: String, vararg args: Any?, tag: String = "Blueteeth") {
        blueteethLogger?.info(String.format("$tag: $message", args))
    }

    fun w(message: String, vararg args: Any?, tag: String = "Blueteeth") {
        blueteethLogger?.warning(String.format("$tag: $message", args))
    }

    fun e(message: String, vararg args: Any?, tag: String = "Blueteeth") {
        blueteethLogger?.error(String.format("$tag: $message", args))
    }

}

File: test-peripheral/Sources/Peripheral/logger.swift
//
//  logger.swift
//  
//
//  Created by SJ on 2020-03-27.
//

import Foundation
import SwiftyTooth

final class MyLogger: Logger {
    func verbose(_ message: String) {
        print("V: " + message)
    }
    func debug(_ message: String) {
        print("D: " + message)
    }
    func info(_ message: String) {
        print("I: " + message)
    }
    func warning(_ message: String) {
        print("W: " + message)
    }
    func error(_ message: String) {
        print("E: " + message)
    }
}
